Upgrade to mui v6

FranckLecuyer opened the pull request:
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines  
- [ ] Tests for the changes have been added (for bug fixes / features)  
- [ ] Docs have been added / updated (for bug fixes / features)  
- [ ] A PR or issue has been opened in all impacted repositories (if any)  

**Does this PR already have an issue describing the problem?**  
<!-- If so, link to this issue using `'Fixes #XXX'` and skip the rest -->
No

**What kind of change does this PR introduce?**  
Upgrade to Material UI (MUI) v6.

**What is the current behavior?**  
The library is based on MUI v5.

**What is the new behavior (if this is a feature change)?**  
The library now depends on MUI v6.

**Does this PR introduce a breaking change or deprecate an API?**  
- [x] Yes  
- [ ] No  

**What changes might users need to make in their application due to this PR? (migration steps)**  

Users must upgrade their application from MUI v5 to MUI v6 to remain compatible with this library.

In particular:
- Update `@mui/material`, `@mui/system`, and related packages to v6
- Ensure no MUI v5 packages remain in the dependency tree

Refer to the official migration guide:  
https://mui.com/material-ui/migration/migration-v5-to-v6/

antoinebhs commented on package-lock.json:
```diff
@@ -67,19 +67,23 @@
             },
             "peerDependencies": {
                 "@emotion/styled": "^11.14.0",
-                "@mui/icons-material": "^5.16.14",
```
too many changes in package lock?

FranckLecuyer committed:
Upgrade to MUI v6

Signed-off-by: Franck LECUYER <franck.lecuyer@rte-france.com>
Signed-off-by: BOUHOURS Antoine <antoine.bouhours@rte-france.com>

# Conflicts:
#	package-lock.json
#	package.json

antoinebhs committed:
Run at the root of this repo
- nvm use
- npm i with mui changes
- yields this package lock, much closer to the one on main

Signed-off-by: BOUHOURS Antoine <antoine.bouhours@rte-france.com>
